runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -4788,341 +4800,332 @@ J9::Z::TreeEvaluator::anewArrayEvaluator(TR::Node * node, TR::CodeGenerator * cg
+      cursor = generateRILInstruction(cg, TR::InstOpCode::NILF, node, sizeReg, 0xff, cursor);
+      cursor = generateRXInstruction(cg, TR::InstOpCode::LA, node, sizeReg, generateS390MemoryReference(sizeReg, resultReg, 0, cg), cursor);
+      //sizeReg has the address of the last initialized byte.
+      cursor = genMemoryZeroingLoop(cg, node, cursor, sizeReg, scratchReg, 1);
+      cursor = generateS390LabelInstruction(cg, TR::InstOpCode::label, node, memoryInitializationEndLabel, cursor);
+      }
+
+   /********************************************* First dimesion class and length *********************************************/
+   // Load first dim length in lower 32bits and second dim length in higher 32bits.
+   cursor = generateRXInstruction(cg, TR::InstOpCode::LG, node, scratchReg, generateS390MemoryReference(dimsPtrReg, 0, cg), cursor);
+   iComment("Start first dim alloc");
+   // Fist dim class and length:
+   bool useCompRefs = TR::Compiler->om.compressObjectReferences();
+   cursor = generateRXInstruction(cg, useCompRefs ? TR::InstOpCode::ST : TR::InstOpCode::STG, node, classReg,
+      generateS390MemoryReference(resultReg, static_cast<int32_t>(TR::Compiler->om.offsetOfObjectVftField()), cg), cursor);
+   cursor = generateRXInstruction(cg, TR::InstOpCode::ST, node, scratchReg, generateS390MemoryReference(resultReg,
+      static_cast<int32_t>(TR::Compiler->om.offsetOfContiguousArraySizeField()), cg), cursor);
+
+   /********************************************* Add leaf arrays (Second dimension) *********************************************/
+   // If number of leaf arrays is zero, jump to the end.
+   TR::LabelSymbol *controlFlowEndLabel = generateLabelSymbol(cg);
+   controlFlowEndLabel->setEndInternalControlFlow();
+   cursor = generateS390CompareAndBranchInstruction(cg, TR::InstOpCode::CL, node, scratchReg, 0, TR::InstOpCode::COND_BE, controlFlowEndLabel, false /* needsCC */, cursor);
+
+   cursor = generateRXInstruction(cg, TR::InstOpCode::LA, node, dim1SizeReg, generateS390MemoryReference(dim1SizeReg, resultReg, 0, cg), cursor);
+   iComment("dim1SizeReg points to the fist leaf array.");
+   // Load component class to class register. In case of comp refs, the calss is 32bits long so using this register for leaf array length value as well.
+   // ClassReg when non-comp refs: |     class      |
+   // ClassReg when comp refs:     | class | length |
+   cursor = generateRXInstruction(cg, TR::InstOpCode::LG, node, classReg, generateS390MemoryReference(classReg, offsetof(J9ArrayClass, componentType) + (useCompRefs ? 4 : 0), cg), cursor);
+   int32_t shiftAmount = TR::Compiler->om.compressedReferenceShift();
+   if (useCompRefs)
+      {
+      // Use the class register for both class and length fields.
+      cursor = generateRXInstruction(cg, TR::InstOpCode::L, node, classReg, generateS390MemoryReference(dimsPtrReg, 0, cg), cursor);
+      if (shiftAmount > 0)
+         {
+         // Keep a compressed dim2 size in higher 32 bits.
```

Comment:
I do not think you addressed my previous comment, what is compressed dim2 size? `Keep a compressed dim2 size in higher 32-bits` has completely different meaning then `For compressedref, dim2 size in higher 32-bits`

Reply:
It is not termed as `compressed` size. You are shifting the size not compressing it.